src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/CoursesControllerTest.kt

```diff
@@ -17,12 +23,16 @@ import java.util.UUID
 @Import(JwtAuthHelper::class)
 class CoursesControllerTest(
   @Autowired val webTestClient: WebTestClient,
-  @Autowired val coursesService: CourseService,
+  @Autowired val inMemoryCourseRepository: InMemoryCourseRepository,
```

Comment:
I think this change (and all changes to this file other than the new test for `coursesPut`) should come in its own commit, or at the very least we should document why it's changed in this commit if it's necessary for the Put endpoint - I don't know if it's directly related. @bell-pa what's the reason for this change?

Reply:
CourseControllerTest used to use an injected CourseSerivce which in turn used InMemoryCourseRepository as a source of data. The controller now calls a service method that takes a parameter, and the method implementation is not trivial. I felt that it was better at this stage to switch to using a mock implementation of CourseService instead with the data for the mock responses being programmed from InMemoryRepository as before.